tests/alignment_check/mod.rs

```diff
@@ -9,30 +9,49 @@ fn unaligned_slicing_integers() {
 
     assert_eq!(transmute_many_permissive::<u16>(bytes), Ok(words.as_ref()));
     assert_eq!(transmute_many_permissive::<u16>(&bytes[1..]),
-               Err(Error::Unaligned(UnalignedError { offset: 1 })));
+               Err(Error::Unaligned(UnalignedError::new(1, &bytes[1..]))));
     assert_eq!(transmute_many_permissive::<u16>(&bytes[2..]), Ok(&words[1..]));
     assert_eq!(transmute_many_permissive::<u16>(&bytes[3..]),
-               Err(Error::Unaligned(UnalignedError { offset: 1 })));
+               Err(Error::Unaligned(UnalignedError::new(1, &bytes[3..]))));
 
     let words = [0x02EE_01FF, 0x04CC_03DD, 0x06AA_05BB];
     let bytes = transmute_to_bytes(&words);
 
     assert_eq!(transmute_many_permissive::<u32>(bytes), Ok(words.as_ref()));
     assert_eq!(transmute_many_permissive::<u32>(&bytes[1..]),
-               Err(Error::Unaligned(UnalignedError { offset: 3 })));
+               Err(Error::Unaligned(UnalignedError::new(3, &bytes[1..]))));
     assert_eq!(transmute_many_permissive::<u32>(&bytes[2..]),
-               Err(Error::Unaligned(UnalignedError { offset: 2 })));
+               Err(Error::Unaligned(UnalignedError::new(2, &bytes[2..]))));
     assert_eq!(transmute_many_permissive::<u32>(&bytes[3..]),
-               Err(Error::Unaligned(UnalignedError { offset: 1 })));
+               Err(Error::Unaligned(UnalignedError::new(1, &bytes[3..]))));
     assert_eq!(transmute_many_permissive::<u32>(&bytes[4..]), Ok(&words[1..]));
     assert_eq!(transmute_many_permissive::<u32>(&bytes[5..]),
-               Err(Error::Unaligned(UnalignedError { offset: 3 })));
+               Err(Error::Unaligned(UnalignedError::new(3, &bytes[5..]))));
 
-    let words = [0x02EE_01FF_04CC_03DD];
+    let words = [0x02EE_01FF_04CC_03DD, 0x06EE_05FF_08CC_07DD];
     let bytes = transmute_to_bytes(&words);
-    assert_eq!(transmute_many_permissive::<u64>(bytes), Ok(words.as_ref()));
+    assert_eq!(transmute_many_permissive::<u64>(bytes), Ok(&words[..]));
+    assert_eq!(transmute_many_permissive::<u64>(&bytes[8..]), Ok(&words[1..]));
     for i in 1..8 {
-        assert_eq!(transmute_many_permissive::<u64>(&bytes[i..]),
-                   Err(Error::Unaligned(UnalignedError { offset: 8 - i })));
+        let outcome = transmute_many_permissive::<u64>(&bytes[i..]);
+        assert_eq!(outcome, Err(Error::Unaligned(UnalignedError::new(8 - i, &bytes[i..]))));
+        #[cfg(feature = "std")]
```

Comment:
Some vertical spacing and a more descriptive name than `number` et al. would help here, this is a tad unreadable ATM :v